### PR TITLE
Update URDF parsing logic and improve R/W operations on files

### PR DESCRIPTION
**Pull request by flferretti**
This PR updates the parsing logic for URDF files by bounding the check on file existence to the maximum path length allowed by the OS. Moreover, some changes were made to make the RW operations on files safer and some final linting was performed. 

Closes https://github.com/ami-iit/comodo/issues/21

----
📚 Documentation preview 📚: https://adam-docs--100.org.readthedocs.build/en/100/

### CODE REVIEW

File: tests/parametric/test_pytorch_computations_parametric.py [hunk 1]
```diff
@@ -27,14 +22,17 @@
...
-robot_file_write = open(model_path, "w")
-robot_file_write.write(robot_urdf_string)
+## Hack to remove the encoding urdf, see https://github.com/icub-tech-iit/ergocub-gazebo-simulations/issues/49
+with open(model_path, "r", encoding="utf-8") as robot_file:
+    robot_urdf_string = (
+        robot_file.read()
+        .replace("<?xml", "")
+        .replace("version='1.0'", "")
+        .replace("encoding='UTF-8'?>", "")
+    )
```
Comment (diegoferigo) [inline tests/parametric/test_pytorch_computations_parametric.py:32]: @traversaro @CarlottaSartore did https://github.com/ros/urdf_parser_py/pull/83 fix https://github.com/ros/urdf_parser_py/issues/82? If yes, maybe these lines that remove the encoding can be removed.

Comment (traversaro) [reply tests/parametric/test_pytorch_computations_parametric.py:32]: I think so, but no release was done, so I guess the issue is still present in the latest release, and the problem is also that the pypi package is abandoned.

Comment (diegoferigo) [reply tests/parametric/test_pytorch_computations_parametric.py:32]: Ok I see, so probably the solution for ADAM is to just migrate to something else, at one point in the future. Thanks for the feedback!

File: examples/mpc-ik.ipynb [hunk 2]
```diff
@@ -90,22 +87,22 @@
...
       "source": [
         "# load scene from xml\n",
-        "model = mujoco.MjModel.from_xml_path(\"mujoco_menagerie/franka_emika_panda/scene.xml\")\n"
+        "model = mujoco.MjModel.from_xml_path(\"mujoco_menagerie/franka_emika_panda/scene.xml\")"
       ]
     },
     {
       "cell_type": "markdown",
+      "metadata": {
+        "id": "CZMO7PsmKUB6"
```
Comment (Giulero) [inline examples/mpc-ik.ipynb:96]: Just selecting a few lines, but I'm considering the whole file. I was wondering: which is the automatic formatter you used? I will put it in the dependencies ;)
I would remove this file from this PR for now (just to enhance the clarity of this PR) and maybe add it in a new PR that just formats the files!

**Comment by Giulero** [comment]
Thanks a lot @flferretti! :) 
I just left a comment related to the automatic format of a file. Automatic sorting of imports is fine. I should have inserted it before tbh :D

**Review by Giulero — APPROVED** [review]

File: examples/mpc-ik.ipynb (hunk 2 above)
Comment (flferretti) [reply examples/mpc-ik.ipynb:96]: I'm using `black[jupyter]` and `isort` for formatting.

Formatting removed in 79f6d67 with rebase

**Comment by flferretti** [comment]
Thank you for the review @Giulero! I've addressed your comments. Before merging, please note that this PR is rebased onto #99

**Pull request merged**